packages/cli/test/unit/openapi/schema-model.unit.js

```diff
@@ -374,14 +385,19 @@ describe('schema to model', () => {
           },
           {
             name: 'first-name',
-            signature: "'first-name'?: string;",
+            signature: 'firstName?: string;',
             decoration: "@property({name: 'first-name'})",
           },
           {
             name: 'last-name',
-            signature: "'last-name'?: Name;",
+            signature: 'lastName?: Name;',
             decoration: "@property({name: 'last-name'})",
           },
+          {
+            name: 'profiles',
+            signature: 'profiles?: ProfileId[];',
+            decoration: "@property.array(String, {name: 'profiles'})",
```

Comment:
I am wondering - why are we including `name` in the property metadata? Isn't the decorator inferring the property name automatically? I believe the property name specified in the metadata must match the property name in TypeScript source, otherwise things will break at runtime.

IMO, we should exclude `name` from the generated property metadata.

Feel free to leave such changes out of scope of this pull request.

Reply:
Some of the property names such as `last-name` have to be set in the `@property` decoration. I didn't bother to test if the original name is the same as the TS property.

Reply:
> Some of the property names such as `last-name` have to be set in the `@property` decoration. I didn't bother to test if the original name is the same as the TS property.

So how is this going to work in practice? Let's say the OpenAPI schema describes a property called `last-name`. This means the clients will be sending request body payload like `{"last-name": "Bajtos"}`.

At runtime, this will create object `{'last-name': 'Bajtos'}`.

To allow TypeScript code to access this data, the TypeScript property must be called `last-name` too. If it's changed e.g. to `lastName`, then 
 - TypeScript code won't be able to access the real property value.
 - Values set from TypeScript (e.g. `data.lastName = 'Feng'`) will be ignored by model's `toJSON` function, thus they won't be persisted in the database and serialized to HTTP responses.

> Some of the property names such as last-name have to be set in the @property decoration.

Are you saying that our code in incorrectly inferring model property name from the TypeScript/JavaScript property name passed by TypeScript to `@property`? I.e. that the following won't work?

```ts
@model()
class MyModel extends Entity {
  @property()
  'last-name': string;
}
``

Reply:
The following code should work:
```ts
@model()
class MyModel extends Entity {
  @property()
  'last-name': string;
}
```

But we prefer to support the following:
```ts
@model('my-model')
class MyModel extends Entity {
  @property('last-name')
  lastName: string;
}
```

Mapping between JS/TS and JSON should be allowed. I'll check if our model class constructor and toJSON() supports so.

Reply:
See https://github.com/strongloop/loopback-next/pull/2743